scripts/github_pr/github_pr.rb

```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/ruby
+
+require 'octokit'
+require 'optparse'
+require 'yaml'
+require 'json'
+
+class GithubAPI
+  def get_client
+    client = Octokit::Client.new(:netrc => true)
+    client.auto_paginate = true
+    client.login
+    client
+  end
+
+  def client
+    @@client ||= get_client
+  end
+
+  def current_rate_limit
+    client.rate_limit.remaining
+  end
+end
+
+
+class GithubClient
+  RESULT_MESSAGES = {
+    :success => 'succeeded',
+    :failure => 'failed',
+    :error   => 'has an error',
+    :pending => 'is pending',
+  }
+
+  def initialize(conf = {})
+    @config = conf
+  end
+
+  def organization
+    @config[:organization]
+  end
+
+  def repository
+    @config[:repository]
+  end
+
+  def org_repo
+    "#{organization}/#{repository}"
+  end
+
+  def context
+    @config[:context]
+  end
+
+  def set_status(commit, details)
```

Comment:
The method names don't follow the ruby conventions; `get_`, `set_`, `is_` should not be used. See https://github.com/bbatsov/ruby-style-guide#naming.